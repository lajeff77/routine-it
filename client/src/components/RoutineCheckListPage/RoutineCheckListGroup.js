import React, {useState} from "react";
import {Box, Tab, Tabs } from "@mui/material";
import RoutineCheckList from "./RoutineCheckList";

const RoutineCheckListGroup = (props) => {

    const [tabIndex, setTabIndex] = useState(0);

    const handleTabChange = (event, newTabIndex) => {
        
        setTabIndex(newTabIndex);
    };

    return(
        <Box>
            <Tabs value={tabIndex}  variant="fullWidth" onChange={handleTabChange} sx={{margin: '30px'}}>
                <Tab label="High Effort" />
                <Tab label="Medium Effort" />
                <Tab label="Low Effort" />
            </Tabs>
            <Box sx={{ padding: 2 }}>
                {tabIndex === 0 && <RoutineCheckList  name="High Effort Routine" tasks={props.highEffortTasks}/>}
                {tabIndex === 1 && <RoutineCheckList  name="Medium Effort Routine" tasks={props.medEffortTasks}/>}
                {tabIndex === 2 && <RoutineCheckList  name="Low Effort Routine" tasks={props.lowEffortTasks}/>}
            </Box>
        </Box>
    )
}

export default RoutineCheckListGroup;