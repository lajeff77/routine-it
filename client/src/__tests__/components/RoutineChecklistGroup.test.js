import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import RoutineCheckListGroup from "../../components/RoutineCheckListPage/RoutineCheckListGroup";

const testProps = {
    routineId:1,
    highEffortTasks:[
        {
            "id": 1,
            "routineId": 1,
            "name": "Brush teeth",
            "highEffort": true,
            "mediumEffort": true,
            "lowEffort": true,
            "complete": false
        },
        {
            "id": 2,
            "routineId": 1,
            "name": "Wash face",
            "highEffort": true,
            "mediumEffort": true,
            "lowEffort": true,
            "complete": false
        },
        {
            "id": 3,
            "routineId": 1,
            "name": "Mirror affirmations",
            "highEffort": true,
            "mediumEffort": true,
            "lowEffort": false,
            "complete": false
        },
        {
            "id": 4,
            "routineId": 1,
            "name": "Do makeup",
            "highEffort": true,
            "mediumEffort": false,
            "lowEffort": false,
            "complete": false
        }
    ],
    medEffortTasks:[ {
        "id": 1,
        "routineId": 1,
        "name": "Brush teeth",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    },
    {
        "id": 2,
        "routineId": 1,
        "name": "Wash face",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    },
    {
        "id": 3,
        "routineId": 1,
        "name": "Mirror affirmations",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": false,
        "complete": false
    }],
    lowEffortTasks:[{
        "id": 1,
        "routineId": 1,
        "name": "Brush teeth",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    },
    {
        "id": 2,
        "routineId": 1,
        "name": "Wash face",
        "highEffort": true,
        "mediumEffort": true,
        "lowEffort": true,
        "complete": false
    }]
}

describe(RoutineCheckListGroup.name, ()=>{
    test('RoutineCheckListGroup matches snapshot', () => {
        const {container} = render(<RoutineCheckListGroup routineId={testProps.routineId} highEffortTasks={testProps.highEffortTasks} medEffortTasks={testProps.medEffortTasks}q lowEffortTasks={testProps.lowEffortTasks}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});