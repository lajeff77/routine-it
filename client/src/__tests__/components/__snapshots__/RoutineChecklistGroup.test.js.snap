// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutineCheckListGroup RoutineCheckListGroup matches snapshot 1`] = `
<div
  class="MuiBox-root css-0"
>
  <div
    class="MuiTabs-root css-qt3m9d-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1ujykiq-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="0"
          type="button"
        >
          High Effort
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1ujykiq-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Medium Effort
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1ujykiq-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Low Effort
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
  <div
    class="MuiBox-root css-19midj6"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-icizyq-MuiTypography-root"
      >
        High Effort Routine
      </h2>
      <fieldset
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <legend
          class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
        >
          Tasks
        </legend>
        <div
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="Brush teeth"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Brush teeth
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="Wash face"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Wash face
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="Mirror affirmations"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Mirror affirmations
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="Do makeup"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Do makeup
            </span>
          </label>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;
